### ISSUE TITLE
Expose LoggerService for custom implementation

### ISSUE THREAD

**Issue by nickdodd79** [damienbod/angular-auth-oidc-client#487]
Hi,

We use a third-party provider to process and report on the logs being produced by our apps. However, there is currently no way to hook into the logging used in this library. What would be ideal is the ability to register a custom provider for the `LoggerService` type, with a custom implementation.

```
{
  provide: LoggerService,
  useClass: AuthLogger
}
```

As the `LoggerService` is not exposed via the public API, this is not possible. I have also tried the full path the service, but get build errors:

```
import { LoggerService } from 'angular-auth-oidc-client/lib/services/oidc.logger.service';
```

Could the service be re-exported as part of the public API?

Thanks.
Nick.

**Comment by damienbod** [comment]
@nickdodd79 

Yes, this sounds like a good idea. Would you like to do a PR?

Greetings Damien

**Comment by nickdodd79** [comment]
Sure, I'll see what I can do 👍

**Comment by damienbod** [comment]
released version 10.0.4, thanks
